ENG-2465 - Add New Login API responses and fields (#3603)
FusionAuth/fusionauth-site: astro/src/content/docs/apis/_user-login-response.mdx

import APIField from 'src/components/api/APIField.astro';
import InlineField from 'src/components/InlineField.astro';

> <APIField name="emailVerificationId" type="String">

Author: leaving out the nuance on both of these, that the tenant has to be set to formfield when the login occurs AND the identifier needs to have an OTP on it (identifier could have an OTP on it but strategy has since been changed.)

Author: That's right. Not sure why it was missed before.

>   An email verification Id. When present, this will represent the user's current email verification Id.

Reviewer: ```suggestion
  An email verification Id, which is needed when performing your own email verification using the `/api/user/verify-email` API.
```

Author: Done

  This field is only returned when the <InlineField>emailConfiguration.verificationStrategy</InlineField> is set to `FormField`.
</APIField>

<APIField name="identityVerificationId" type="String" since="1.99.9">

>   An identity verification Id.

Reviewer: ```suggestion
  An identity verification Id, which is needed when performing your own identity verification using the `XXXXXXXX` API.
```

Author: Done

  * A phone number - the <InlineField>phoneConfiguration.verificationStrategy</InlineField> is set to `FormField`.
</APIField>

<APIField name="pendingIdPLinkId" type="String" since="1.28.0" renderif={!!props.idp_response}>
  The pending identity provider link Id. This value is created when logging in with an identity provider configured with a linking strategy of `Create a pending link`. It will only be included in the response body when this strategy is configured and a link does not yet exist for the user. It is used in conjunction with the [Link APIs](/docs/apis/identity-providers/links) to complete a pending link.
</APIField>

<APIField name="refreshToken" type="String">
  The refresh token that can be used to obtain a new access token once the provided one has expired.

  Because a refresh token is per user and per application, this value will only be returned when an <InlineField>applicationId</InlineField> was provided on the login request and the user is registered to the application.

  You must explicitly allow generation of refresh tokens when using the Login API.

>   Configure the <InlineField>application.loginConfiguration.generateRefreshTokens</InlineField> setting via the API or enable the setting by navigating to the **Application -> My Application -> Security** tab.

Reviewer: Should be using `<Breadcrumb>foo -> bar -> baz</Breadcrumb>`

Author: Done

import APIField from 'src/components/api/APIField.astro';
import Breadcrumb from 'src/components/Breadcrumb.astro';
import InlineField from 'src/components/InlineField.astro';

<APIField name="emailVerificationId" type="String">
  An email verification Id, which is needed when performing your own email verification using the `/api/user/verify-email` API.

>   This field is only returned when the <InlineField>emailConfiguration.verificationStrategy</InlineField> is set to `FormField`.

Reviewer 2: ```suggestion
  This field is only returned when the <InlineField>tenant.emailConfiguration.verificationStrategy</InlineField> is set to `FormField`.
```

Author: Fixed

</APIField>

<APIField name="identityVerificationId" type="String" since="1.99.9">
  An identity verification Id, which is needed when performing your own identity verification using the `/api/identity/verify` API.

  This field is only returned under the following conditions, based on the identity that was used to log in:

>   * An email address - the <InlineField>emailConfiguration.verificationStrategy</InlineField> is set to `FormField`.
>   * A phone number - the <InlineField>phoneConfiguration.verificationStrategy</InlineField> is set to `FormField`.

Reviewer 2: ```suggestion
  * An email address - the <InlineField>tenant.emailConfiguration.verificationStrategy</InlineField> is set to `FormField`.
  * A phone number - the <InlineField>tenant.phoneConfiguration.verificationStrategy</InlineField> is set to `FormField`.
```

Author: Fixed
